packages/pigeon/lib/objc_generator.dart

```diff
@@ -427,7 +431,9 @@ void _writeHostApiDeclaration(Indent indent, Api api, ObjcOptions options) {
       lastArgType = 'FlutterError *_Nullable *_Nonnull';
       lastArgName = 'error';
     }
-    if (!func.returnType.isNullable) {
+    if (!func.returnType.isNullable &&
+        !func.returnType.isVoid &&
+        !func.isAsynchronous) {
```

Comment:
Here too.

Reply:
This is a return type so it can be void.